messagix/client.go

```diff
@@ -388,3 +393,28 @@ func (c *Client) GetTaskId() int {
 	c.activeTasks = append(c.activeTasks, id)
 	return id
 }
+
+func (c *Client) EnableSendingMessages() {
+	c.sendMessagesCond.L.Lock()
+	c.canSendMessages = true
+	c.sendMessagesCond.Broadcast()
+	c.sendMessagesCond.L.Unlock()
+}
+
+func (c *Client) WaitUntilCanSendMessages(timeout time.Duration) error {
+	c.sendMessagesCond.L.Lock()
+	defer c.sendMessagesCond.L.Unlock()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for !c.canSendMessages {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("timeout waiting for sending messages")
+		default:
+			c.sendMessagesCond.Wait()
```

Comment:
This comment made realize that the error returned by `WaitUntilCanSendMessages` wasn't being properly handed. I've fixed that.

> I think this will still block until it's reconnected

However I can't see the scenario you're describing, could you elabore on that one? Thanks!

Reply:
Good catch, indeed the timeout was not being triggered. I've proposed a solution for this in the comment below.